src/bandersnatch/utils.py

```diff
@@ -89,7 +94,7 @@ def update_safe(filename, **kw):
             prefix=os.path.basename(filename) + '.', **kw) as tf:
         if os.path.exists(filename):
             os.chmod(tf.name, os.stat(filename).st_mode & 0o7777)
-        tf.has_changed = False
+        tf.has_changed = False  # type: ignore
```

Comment:
What's the error here? I feel we should be able to fix it.

Reply:
`has_changed` is not an attribute of file objects, there's attribute stuffing happening here